### PR TITLE
Reporting period overlap fix

### PR DESCRIPTION
**Pull request by bamader**
## Background
Users were experiencing a selection of bugs that either blocked them from submitting accurate data or confused them when it came to submitting revision requests. This PR fixes those pain points.

## GitHub Issue
https://github.com/ctoec/data-collection/issues/1161

## Validation Plan
Tested enrollment parsing/EditRecord view with children who had first and last reporting periods _not_ in the same month as well as children who had these fields in the same month. Also checked that school readiness checkboxes were appropriate pre-checking when a user opened the revision request form.

## Automated Testing
- [ ] All unit tests are passing.
- [ ] All e2e tests are passing.
- [ ] If applicable, unit tests have been added to cover this changeset (not applicable).
- [ ] If applicable, e2e tests have been added to cover this changeset (not applicable).

### CODE REVIEW

File: client/src/containers/Home/RevisionRequest.tsx [hunk 1]
```diff
@@ -82,12 +82,11 @@ export const RevisionRequest: React.FC = () => {
       apiGet('funding-spaces', accessToken)
         .then((res: FundingSpace[]) => {
           res.forEach((fs) => {
-            const stringRep =
-              fs.ageGroup +
-              ' - ' +
-              fs.source.split('-')[1].trim() +
-              ' - ' +
-              fs.time;
+            let spaceName = fs.source.split('-')[1].trim();
```
Comment (jlhogan) [inline client/src/containers/Home/RevisionRequest.tsx:85]: I updated this ticket to havesome other changes; mind separating these out into two separate PRs so we can merge the funding validation bug and fully resolve all of the revision request stuff?

Comment (bamader) [reply client/src/containers/Home/RevisionRequest.tsx:85]: Sure can do. I'll leave this one as the reporting period validation and move the revision stuff elsewhere.

**Review by JordanGuinn — APPROVED** [review]
HEH, my kinda changeset right here 😁😁😁😁😁😁

**Review by jlhogan — APPROVED** [review]
wahoooooo

**Comment by bamader** [comment]
Do y'all want to do your own validations/testing in QA for this, or am I good to just run the qa tests and then merge?

**Pull request merged**